### PR TITLE
Support for setting EtherType of EthernetHeader in ATE traffic

### PR DESCRIPTION
**Pull request by mingyangcisco**
This pull request is regarding the issue  https://github.com/openconfig/ondatra/issues/22

The changes include:

adding ethernet_type field in the ate.proto 
adding WithEthernetType function in headers.go
enabling setting ethernet-type in stacks.go and also adding related to test in stacks_test.go

Example Usage:
		ethHeader := ondatra.NewEthernetHeader()
		ethHeader.WithSrcAddress("00:11:01:00:00:01")
		ethHeader.WithDstAddress("00:22:01:00:00:01")
		ethHeader.WithEthernetType("0x88cc")

### CODE REVIEW

File: headers.go [hunk 1]
```diff
@@ -75,6 +75,12 @@ func (h *EthernetHeader) WithVLANID(vid uint16) *EthernetHeader {
 	return h
 }
 
+// WithEthernetType sets string type Ethernet-Type of the Ethernet header to the specified value
+func (h *EthernetHeader) WithEthernetType(ethernetType string) *EthernetHeader {
```
Comment (greg-dennis) [inline headers.go:79]: to be consistent with the spec, let's call this property "etherType" throughout and make it an "int" type

**Review by greg-dennis — CHANGES_REQUESTED** [review]

File: headers.go (hunk 1 above)
Comment (mingyangcisco) [reply headers.go]: thanks for checking. I made the changes as suggested.

File: internal/ate/stacks_test.go [hunk 2]
```diff
@@ -154,6 +155,41 @@ func TestHeaderStacks(t *testing.T) {
 				},
 			}},
 		},
+	}, {
+		desc: "ethernet header w/ ethernet-type",
```
Comment (greg-dennis) [inline internal/ate/stacks_test.go:159]: Can you just add an etherType field to the "ethernet header" case instead. In fact, if you could also do a bit of cleanup and get rid of the "ethernet header w/ vlan id" and do the same thing there (just add the vlan id to the header), that would be appreciated.

**Review by greg-dennis — CHANGES_REQUESTED** [review]

File: headers.go [hunk 3]
```diff
@@ -75,6 +75,12 @@ func (h *EthernetHeader) WithVLANID(vid uint16) *EthernetHeader {
 	return h
 }
 
+// WithEtherType sets string type EtherType of the Ethernet header to the specified value
```
[the remark this answers is not in the document]
Comment (mingyangcisco) [reply headers.go]: thanks for catching this. changes are done now.

File: internal/ate/stacks_test.go (hunk 2 above)
Comment (mingyangcisco) [reply internal/ate/stacks_test.go]: sure. merged those 2 sub testcases as suggested. thanks

**Comment by greg-dennis** [comment]
Thank you. This is ready for our internal review process now. When that is done and submitted, I will push the final code back out.

**Comment by mingyangcisco** [comment]
> Thank you. This is ready for our internal review process now. When that is done and submitted, I will push the final code back out.

Cool. Thanks

**Comment by greg-dennis** [comment]
This change has been incorporated.

**Pull request closed without merging**